src/BlockTypes/ProductSummary.php

```diff
@@ -51,5 +53,67 @@ protected function get_block_type_supports() {
 	protected function register_block_type_assets() {
 		parent::register_block_type_assets();
 		$this->register_chunk_translations( [ $this->block_name ] );
+		return null;
+	}
+
+	/**
+	 * Register the context.
+	 */
+	protected function get_block_type_uses_context() {
+		return [ 'query', 'queryId', 'postId' ];
+	}
+
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		if ( ! empty( $content ) ) {
+			parent::register_block_type_assets();
+			$this->register_chunk_translations( [ $this->block_name ] );
+			return $content;
+		}
+
+		$post_id         = $block->context['postId'];
+		$product         = wc_get_product( $post_id );
+		$summary_content = self::get_summary_content( $product );
+
+		if ( ! $summary_content ) {
+			return null;
+		}
+
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
+		$classname          = isset( $attributes['className'] ) ? $attributes['className'] : '';
+
+		$output  = '<div class="wc-block-components-product-summary '
+			. esc_attr( $classes_and_styles['classes'] ) . ' ' . esc_attr( $classname ) . '"'
+			. isset( $classes_and_styles['styles'] ) ? ' style="' . esc_attr( $classes_and_styles['styles'] ) . '"' : '';
```

Comment:
The tertiary like this doesn't seem to work well for me (sorry, I know if was my original suggestion). :sweat: 

![imatge](https://user-images.githubusercontent.com/3616980/204796431-3df490a7-331c-43e8-aa1d-93851e4ab736.png)

But I think it you do it as a normal if, it works well, then:

```PHP
if ( isset( $classes_and_styles['styles'] ) ) {
	$output .= ' style="' . esc_attr( $classes_and_styles['styles'] ) . '"';
}
```

Reply:
Oh, I did test the PR after making the changes, but it looks like my changes weren't reflected correctly while I was doing the testing. Thanks for pointing this out. 🙂